code/game/machinery/cloning.dm

```diff
@@ -308,7 +308,7 @@
 		return
 
 	if((occupant) && (occupant.loc == src))
-		if((occupant.stat == DEAD) || (occupant.suiciding) || !occupant.key)  //Autoeject corpses and suiciding dudes.
+		if((occupant.stat == DEAD) || (occupant.suiciding) || !occupant.key || occupant.hellbound)  //Autoeject corpses and suiciding dudes.
 			locked = 0

```

Comment:
This won't be enough to completely prevent cloning - this body already has their key and mind linked, so they can be put in a cryo cell, and survive - or if they don't, a defib will still bring them back.